mitmproxy/proxy/layer.py

```diff
@@ -111,7 +121,34 @@ def handle_event(self, event: events.Event) -> CommandGenerator[None]:
             if self.debug is not None:
                 yield self.__debug(f">> {event}")
             command_generator = self._handle_event(event)
-            yield from self.__process(command_generator)
+            send = None
+
+            # inlined copy of __process to reduce call stack.
+            # <✂✂✂>
+            try:
+                command = command_generator.send(send)
+            except StopIteration:
+                return
+
+            while True:
+                if self.debug is not None:
+                    if not isinstance(command, commands.Log):
+                        yield self.__debug(f"<< {command}")
+                if command.blocking is True:
+                    command.blocking = self  # assign to our layer so that higher layers don't block.
```

Comment:
This just threw me off. Could you quickly explain why this is `blocking: Union[bool, "mitmproxy.proxy.layer.Layer"]`? I've seen at least one occurrence of a truthy check `if command.blocking:` but no place where `blocking` is actually used as `Layer`?

Reply:
@sanlengjingvv also mentioned that, I expanded the documentation a bit. :)